### DISCUSSION TITLE
How to abort fzf while retaining input?

### DISCUSSION THREAD

**Discussion by totoroot** [junegunn/fzf#3612] [Q&A]
I use fzf, among other things, for searching for commands in my shell history and it's been great.

However, one thing I've never managed to configure, is to abort the search if I didn't find the command I was looking for, while still retaining input.

Preferably I would like to be able to retain what I've put in so far if there was no match, once I hit <Enter> and if there was a match, but I'm not happy with it, abort fzf with some key combination like `<Ctrl> + <C>`and retain whatever I already wrote.

Currently in both cases the input just gets cleared once I abort fzf with `<Ctrl> + <C>`.

How should I go about configuring this? Any pointers towards a solution are welcome.

**Comment by LangLangBart** [comment]
Does the following work for you ?

```bash
export FZF_CTRL_R_OPTS='--bind "ctrl-c:transform:[[ $FZF_MATCH_COUNT -eq 0 ]] && echo print-query || echo abort"'
```

The `fzf` tool exports the environment variable `FZF_MATCH_COUNT` starting from version
_0.46.0_[^1]. We use this variable to check if there are any matches. If there aren't any, we print
the query. Otherwise, we abort the action.

The `transform` action was introduced in version _0.45.0_[^2]. It allows the user to evaluate a
condition and then perform either action A or action B, depending on the result.

---

### Alternatives

Issue report #3528 is somewhat related to this topic. You might find it helpful, so please take a look.

[^1]: [CHANGELOG.md 0.46.0](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md#0460)
[^2]: [CHANGELOG.md 0.45.0](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md#0450)
